cmd/config/etcd/etcd.go

```diff
@@ -124,6 +139,8 @@ func lookupLegacyConfig(rootCAs *x509.CertPool) (Config, error) {
 	cfg.DialKeepAliveTime = defaultDialKeepAlive
 	cfg.Endpoints = etcdEndpoints
 	cfg.CoreDNSPath = "/skydns"
+	// default path prefix for all keys on etcd, other than CoreDNSPath.
+	cfg.PathPrefix = ""
```

Comment:
This method always runs on start and I can't reassign a prefix value.

Reply:
you need to use `mc admin config set`

Reply:
or use MINIO_ETCD_PATH_PREFIX

Reply:
Hmmm... But I can use `mc admin config set` only after started a MinIO. But when MinIO started it already creates many keys on etcd without prefix. And it's no good. 😞 
I want to set the environment variable `MINIO_ETCD_PATH_PREFIX` for the first start MinIO for safe create all keys on etcd with a custom prefix. 😄

Reply:
> or use MINIO_ETCD_PATH_PREFIX

Yep, I use it. But it does not work because MinIO always calls `lookupLegacyConfig` and set a default empty prefix.

Reply:
It won't if you are using legacy ENV @ihostage you need to make sure to set _STATE=on as well

Reply:
Done shall be fixed @ihostage